src/Model/Item.php

```diff
@@ -3044,7 +3044,12 @@ public static function prepareBody(array &$item, bool $attach = false, bool $is_
 		} elseif (empty($shared_item['uri-id']) && empty($item['quote-uri-id']) && ($item['network'] != Protocol::DIASPORA)) {
 			$media = Post\Media::getByURIId($item['uri-id'], [Post\Media::ACTIVITY]);
 			if (!empty($media)) {
-				$shared_item = Post::selectFirst($fields, ['plink' => $media[0]['url'], 'uid' => [$item['uid'], 0]]);
+				$shared_item = Post::selectFirst($fields, ['uri-id' => $media[0]['media-uri-id'], 'uid' => [$item['uid'], 0]]);
+				if (empty($shared_item['uri-id'])) {
+					$shared_item = Post::selectFirst($fields, ['plink' => $media[0]['url'], 'uid' => [$item['uid'], 0]]);
+				} elseif (strtolower($shared['post']['uri']) != strtolower($media[0]['url'])) {
```

Comment:
@annando - this leads to:
```console
friendica-app-1  | Friendica[146]: index [WARNING]: E_WARNING: Undefined array key "post" {"code":2,"message":"Undefined array key \"post\"","file":"/var/www/html/src/Model/Item.php","line":3050} - {"file":"Item.php","line":3050,"function":"prepareBody","request-id":"e03ade7c15a90688af4b9ad68e97cf91","uid":"8ed524"}
friendica-app-1  | Friendica[146]: index [WARNING]: E_WARNING: Trying to access array offset on value of type null {"code":2,"message":"Trying to access array offset on value of type null","file":"/var/www/html/src/Model/Item.php","line":3050} - {"file":"Item.php","line":3050,"function":"prepareBody","request-id":"e03ade7c15a90688af4b9ad68e97cf91","uid":"8ed524"}
```

But I don't understand how `$shared['post']` can be anything than null, because this is the `elseif` to the the `if (!empty($shared['post']))`

Reply:
ah that would be right ,wouldn't it:
```suggestion
				} elseif (strtolower($shared_item['post']['uri']) != strtolower($media[0]['url'])) {
```

Reply:
Yeah, that line has to be changed to `} elseif (!in_array(strtolower($media[0]['url']), $shared_links)) {`.